src/bin/common/parse_command.ml

```diff
@@ -997,22 +997,46 @@ let parse_output_opt =
     Term.(const not $ no_tableaux_cdcl_in_theories)
   in
 
+  let optim =
+    let doc =
+      "Enable model optimization (experimental)."
+    in
+    Arg.(value & flag & info ["optimize"] ~doc ~docs:s_models)
+  in
+
   let set_sat_options =
-    let set_sat_options sat_solver cdcl_tableaux_inst cdcl_tableaux_th =
-      set_sat_solver sat_solver;
+    let set_sat_options sat_solver cdcl_tableaux_inst cdcl_tableaux_th optim =
+      set_optimize optim;
       begin match sat_solver with
-        | CDCL_Tableaux ->
+        | Util.CDCL_Tableaux ->
+          set_sat_solver sat_solver;
+          set_cdcl_tableaux_inst cdcl_tableaux_inst;
+          set_cdcl_tableaux_th cdcl_tableaux_th
+        | Util.CDCL ->
+          set_sat_solver sat_solver;
+          set_cdcl_tableaux_inst false;
+          set_cdcl_tableaux_th false
+        | s when optim ->
+          (* Optimization is so far only sound for CDCL & CDCL_Tableaux. We set
+             CDCL_Tableaux by default. *)
+          Fmt.pf
+            Format.err_formatter
+            ";Warning: --optimize is not compatible with %a: forcing it to %a@."
+            Util.pp_sat_solver s
+            Util.pp_sat_solver Util.CDCL_Tableaux;
+          set_sat_solver Util.CDCL_Tableaux;
```

Comment:
I don't think overriding an explicitly requested solver is a good thing to do — we should rather fail using `term_result` if conflicting options are given.